mlflow/spark.py

```diff
@@ -58,6 +101,13 @@ def save_model(spark_model, path, mlflow_model=Model(), conda_env=None, jars=Non
             str(type(spark_model))))
     if not isinstance(spark_model, PipelineModel):
         raise Exception("Not a PipelineModel. SparkML can save only PipelineModels.")
+    # Spark ML stores the model on DFS if running on a cluster
+    # Save it to a temp dir first and copy it to local path
+    dfs_tmpdir = _tmp_path(32)
```

Comment:
This is a top-level directory? We should probably hide it away, in something like `/tmp/mlflow/_tmp_path()`. There is definitely a nontrivial probability that this method fails and leaves garbage in the directory, and we shouldn't leave top-level directories around.

Reply:
One thing though - if we make the tmp/mlflow, those will be left on the filesystem for sure.